artichoke-backend/src/extn/core/string/string.rb

```diff
@@ -785,4 +776,67 @@ def valid_encoding?
     # NOOP and is here for compatibility.
     true
   end
+
+  private
```

Comment:
For helper methods like this written in Ruby, I've tended to move them out of the core classes into an `Artichoke` helper module, like this one for `Array#inspect`:

https://github.com/artichoke/artichoke/blob/6b458183f932ef8bf96d5683fe4c5366b948e421/artichoke-backend/src/extn/core/array/array.rb#L3-L11

Can we move these `tr` helper methods into an `Artichoke::String` module located at the top of this source file?

We'd then delegate to these methods in the `String#tr` and friends impls:

https://github.com/artichoke/artichoke/blob/6b458183f932ef8bf96d5683fe4c5366b948e421/artichoke-backend/src/extn/core/array/array.rb#L916-L933

Reply:
Also, as a current implementation limitation of Artichoke, `private` without a method name has no effect. To make a method private, one must call `private :tr_expand_str` after the method definition.